src/content/IntegratedTerminal.tsx

```diff
@@ -12,8 +26,106 @@ const IntegratedTerminal = (props: {
   const { rpcUrl, formProps, simulateWalletPassthrough, strictTokenList, defaultExplorer } = props;
   const [isLoaded, setIsLoaded] = useState(false);
 
-  const passthroughWalletContextState = useWallet();
+  const passthroughWalletContextState = useUnifiedWallet();
   const { setShowModal } = useUnifiedWalletContext();
+  const walletPublicKey = React.useMemo(
+    () => passthroughWalletContextState?.publicKey?.toString(),
+    [passthroughWalletContextState?.publicKey?.toString()],
+  );
+
+  const onRequestIxCallback: IInit['onRequestIxCallback'] = async (ixAndCb) => {
```

Comment:
do take note that Terminal will not be shipped with `@jup-ag/common`, since it's pretty much the integrator's responsibility of crafting/manipulating the IXs, the duty to submitting and parsing tx will fall on the integrator as well